### PR TITLE
Add support for Iceape-UXP

### PR DESCRIPTION
**Pull request by djames1**
Add support for IceApe-UXP (Seamonkey Fork)
https://wiki.hyperbola.info/doku.php?id=en:project:iceape-uxp

### CODE REVIEW

**Comment by JustOff** [comment]
Have you verified that CAA is working there properly?
Also, please specify the correct `minVersion`.

**Comment by djames1** [comment]
@JustOff Yes I am using it on IceApe-UXP.

**Comment by JustOff** [comment]
Btw, is there any simple method to try IceApe-UXP? As far as I can see, Hyperbola live image provides only a boring command line without a GUI, and their wiki does not describe a way to get a working system by running some kind of step-by-step wizard.

**Comment by djames1** [comment]
@JustOff I do not use Hyperbola, I use Gentoo. I have an ebuild available [here](https://github.com/djames1/uxp-overlay/blob/master/www-client/iceape-uxp/iceape-uxp-2.8.ebuild).

The build instructions are:
- Download IceDove-UXP (their Thunderbird 52.x fork) source and extract it
- Download IceApe-UXP source and move it to a folder called `suite` in the `icedove-uxp-version` folder.
- Download latest UXP source and move it to a folder called `mozilla` in the `icedove-uxp-version` folder.
- Create `.mozconfig` in the `icedove-uxp-version` folder.
- Run `mozilla/mach build` from the `icedove-uxp-version` folder.

**Comment by mattatobin** [comment]
JustOff, these are comm applications as in Thunderbird and SeaMonkey not Phoenix-based.

**Comment by JustOff** [comment]
> Yes I am using it on IceApe-UXP.

Well, I managed to get a working `Iceape-UXP` binary (btw, it's not `IceApe-UXP`), and I have serious doubts about your words. In this particular case, adding the new GUID to `install.rdf` is not enough to make the add-on work properly, and it is impossible not to notice this (not to mention that you used the wrong GUID here).

**Comment by mattatobin** [comment]
Well the problem with adding Mail and Suite applications is that while some of the archived extensions might be applicable the entire store is focused on the fact that these are Firefox extensions so most of what is there will never work on non-Phoenix based applications.

IMO this person is simply a moron.

**Comment by mattatobin** [comment]
Well the problem with adding Mail and Suite applications as a target is that while some of the archived extensions might be applicable, the entire store is focused on the fact that these are Firefox extensions so most of what is there will never work on non-Phoenix based applications.

**NOT TO MENTION, both Hyperbola Mail and Suite applications do NOT use the original Mozilla GUIDs and WILL NOT install any of the extensions anyway.**

IMO this person is simply a moron and the request should just be ignored because it hardly makes sense without major changes in focus of CAA as a full on AMO emulation. As well as, "completing" the store to include the Thunderbird and SeaMonkey Extensions and that, as previously indicated, was not in scope for CAA proper.

See: Issue #58

**Comment by JustOff** [comment]
Leaving emotions aside (thanks in advance), I'm not against the possibility of using CAA in Iceape-UXP (as well as in SeaMonkey, see #14), although this really doesn't make much sense, except maybe as a directory. I just want it to be done thoughtfully, responsibly and correctly.

**Comment by djames1** [comment]
@JustOff The addon is working in Iceape-UXP. All I did locally was add the GUID to install.rdf and nothing more. I am aware that the addons in the archive will not install without at minimum downloading them adding the Iceape-UXP GUID to install.rdf, and in many cases they will need additional work beyond that, just as they would with regular SeaMonkey.

Many addons that aren't written with specific intentions to run on SeaMonkey/Iceape-UXP will run just fine with the GUID added and nothing more, two such examples from your GitHub repositories are Ublock Origin Legacy and Cookies Exterminator (I planned to submit a PR for Cookies Exterminator next). I personally think that just having some sort of warning to alert SeaMonkey/Iceape-UXP users that this addon isn't supported on their browser when they browse to `caa:` should suffice. I can add this message if you would like me to do so.

I would rather submit this to the official CAA than have to maintain a fork specific to SeaMonkey and IA-UXP, which is why I submitted this in the first place.

This addon does indeed work properly on Iceape-UXP, see screenshot.
Screenshot_2020-02-27_11-17-25

**Comment by djames1** [comment]
Fixed incorrect GUID that the moron @mattatobin mentioned.

**Comment by JustOff** [comment]
Sorry, but to have a broken button and menu does not mean to have a properly working extension, and you did not mention this when I asked if everything is ok. If you want this pull request to be accepted, then please make all the necessary changes so that everything works correctly. You can find all the key points using [this simple query](https://github.com/JustOff/ca-archive/search?l=JavaScript&q=SeaMonkey&type=Code). Also, pay attention to the application name spelling in the commit description and comments, and to the spaces and tabs used in the code.

_PS: I really do not welcome insults here, regardless of who first started this, and I ask everyone to have respect for this request._

**Comment by djames1** [comment]
@JustOff I have modified the addon to check for Iceape-UXP in the same places that it checks for SeaMonkey. The button in the toolbar is properly working now. I also edited the commit message. I apologize for wasting your time with that, I will do more thorough testing in the future.

File: content/addon.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ let Addon = {
 		let appver;
 		if (Services.appinfo.name == "Pale Moon") {
 			appver = "27.9";
-		} else if (Services.appinfo.name != "SeaMonkey") {
+		} else if (Services.appinfo.name != "SeaMonkey" || Services.appinfo.name != "Iceape-UXP") {
```
Comment (JustOff) [inline content/addon.js]: There should be `&&`, not `||`.

File: content/list.js [hunk 2]
```diff
@@ -182,7 +182,7 @@ let List = {
 			let appver;
 			if (Services.appinfo.name == "Pale Moon") {
 				appver = "27.9";
-			} else if (Services.appinfo.name != "SeaMonkey") {
+			} else if (Services.appinfo.name != "SeaMonkey" || Services.appinfo.name != "Iceape-UXP") {
```
Comment (JustOff) [inline content/list.js]: Ditto.

File: content/versions.js [hunk 3]
```diff
@@ -110,7 +110,7 @@ let Versions = {
 			let appver;
 			if (Services.appinfo.name == "Pale Moon") {
 				appver = "27.9";
-			} else if (Services.appinfo.name != "SeaMonkey") {
+			} else if (Services.appinfo.name != "SeaMonkey" || Services.appinfo.name != "Iceape-UXP") {
```
Comment (JustOff) [inline content/versions.js]: Ditto.

**Comment by djames1** [comment]
@JustOff It appears I might be a moron after all! Fixed.

**Comment by JustOff** [comment]
Well, to be honest it was [really not the neatest](https://github.com/JustOff/ca-archive/commit/bd86c5aa01d54e98026467b0b64f669a6a8140ca) pull request.

**Pull request merged**